### PR TITLE
Add note on sink's eventual consistency

### LINKED ISSUE #1577: Document: feat(stream): backfill resuming for sink
This issue tracks the documentation update needed for the merged PR #13665.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/13665
Source PR Merged At: 2023-11-29T10:13:21Z

### PR DESCRIPTION
**Pull request by kwannoel**
## Info

Update info about sink's eventual consistency.

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/13665

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1577

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/data-delivery.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ RisingWave supports delivering data to downstream systems via its sink connector
 
 To stream data out of RisingWave, you must create a sink. A sink is an external target that you can send data to. Use the [`CREATE SINK`](/sql/commands/sql-create-sink.md) statement to create a sink. You need to specify what data to be exported, the format, and the sink parameters.
 
+Sinks will take some time to catch up to their upstream source(s) due to latency of the sink, connector or backfilling process.
```
Comment (CharlieSYH) [inline docs/data-delivery.md]: We should add context to this note. Consider:

```suggestion
Sinks become visible right after you create them, regardless of the backfilling status. Therefore, it's important to understand that the data in the sinks may not immediately reflect the latest state of their upstream sources due to the latency of the sink, connector, and backfilling process.
```

File: docs/data-delivery.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ RisingWave supports delivering data to downstream systems via its sink connector
...
+So you can expect that the data in the sink will be eventually consistent.
```
Comment (CharlieSYH) [inline docs/data-delivery.md]: "Eventually" is not clear enough. Users might want to know when the sink data can fully align with the source data. Consider:

```suggestion
To determine whether the process is complete and the data in the sink is consistent, refer to [Monitor statement progress](/manage/view-statement-progress.md).
```

Comment (CharlieSYH) [reply docs/data-delivery.md]: Btw, it might be convenient to users if RisingWave could automatically return the status when the process is complete.

**Review by CharlieSYH — APPROVED** [review]
Proposed some suggestions.

**Comment by kwannoel** [comment]
Thanks for improved phrasing @CharlieSYH . LGTM

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks for making the updates. @kwannoel

**Pull request merged**